### ISSUE TITLE
how to format lastmod to localDataTime

### ISSUE THREAD

**Issue by greetfish** [nuxt-community/sitemap-module#219]
what we need is 2021-08-27T15:19:22 +8:00
but we can only got 2021-08-27T15:19:22.625z

`
const routeItem = {
   url: '/' + element.slug + '/',
   lastmod: format(new Date(), 'yyyy-MM-dd') // after date-fns/format got 2021-08-27T00:00:00.000z
}
`

`
       const routeItem = {
          url: '/' + element.slug + '/',
          lastmod: new Date() // 2021-08-27T15:19:22.625z
        }
`

**Comment by NicoPennec** [comment]
According the [sitemap protocol](https://www.sitemaps.org/protocol.html), the `lastmod` date should be in [W3C Datetime](https://www.w3.org/TR/NOTE-datetime) format. 

You can use the native **ISO format** (on GMT timezone) instead of the `fns/format` helper.

eg.
```js
lastmod: new Date().toISOString()
```

**Comment by FairyFanFan** [comment]
> According the [sitemap protocol](https://www.sitemaps.org/protocol.html), the `lastmod` date should be in [W3C Datetime](https://www.w3.org/TR/NOTE-datetime) format.
> 
> You can use the native **ISO format** (on GMT timezone) instead of the `fns/format` helper.
> 
> eg.
> 
> ```js
> lastmod: new Date().toISOString()
> ```

i just wanna set lastmod's value just like '2024-01-01' not iso fomat , can u teach me how to set it ?

**Comment by greetfish** [comment]
你好，你的来信我已收到

**Comment by NicoPennec** [comment]
> ```js
> lastmod: new Date().toISOString()
> ```

As bellow:

```js
lastmod: new Date().toISOString().substring(0, 10)
```

**Comment by FairyFanFan** [comment]
> > ```js
> > lastmod: new Date().toISOString()
> > ```
> 
> As bellow:
> 
> ```js
> lastmod: new Date().toISOString().substring(0, 10)
> ```

thx, but it will be forced to transfer to iso type, cause so

urce code in v2.4.0 are "normaliseDate(e.lastmod)", normalise function force date string transfer to iso type <img width="603" alt="截屏2024-07-29 17 18 33" src="
<img width="650" alt="截屏2024-07-29 17 18 59" src="https://github.com/user-attachments/assets/5
<img width="688" alt="截屏2024-07-29 17 19 18" src="https://github.com/user-attachments/assets/960abdeb-eaa5-4f61-adff-0a43fa278214">
44446da-96fe-437e-8f00-8a660d222011">
https://github.com/user-attachments/assets/f5b30a6c-d76c-4f32-a804-b758d49a5a75">

**Comment by NicoPennec** [comment]
> > > ```js
> > > lastmod: new Date().toISOString()
> > > ```
> > 
> > 
> > As bellow:
> > ```js
> > lastmod: new Date().toISOString().substring(0, 10)
> > ```
> 
> thx, but it will be forced to transfer to iso type, cause so
> 
> urce code in v2.4.0 are "normaliseDate(e.lastmod)", normalise function force date string transfer to iso type <img width="603" alt="截屏2024-07-29 17 18 33" src=" <img width="650" alt="截屏2024-07-29 17 18 59" src="https://github.com/user-attachments/assets/5 <img alt="截屏2024-07-29 17 19 18" width="688" src="https://private-user-images.githubusercontent.com/12137715/352989500-960abdeb-eaa5-4f61-adff-0a43fa278214.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIyNjY4NTksIm5iZiI6MTcyMjI2NjU1OSwicGF0aCI6Ii8xMjEzNzcxNS8zNTI5ODk1MDAtOTYwYWJkZWItZWFhNS00ZjYxLWFkZmYtMGE0M2ZhMjc4MjE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI5VDE1MjIzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMzYjY3NDRmZGMzMmFkZTk1MjdhY2FmNWVjMzQyNTNjMTZkZjc5OGIxZTdlN2IyZmM3Njg2NDk3NDE5NmFiMGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-AV373juotKxexNZdCZD0bV3zf2FGql19cirBIBumA8"> 44446da-96fe-437e-8f00-8a660d222011"> [https://github.com/user-attachments/assets/f5b30a6c-d76c-4f32-a804-b758d49a5a75">](https://github.com/user-attachments/assets/f5b30a6c-d76c-4f32-a804-b758d49a5a75%22%3E)

I don't understand your explanation 🤔 
This typescript code is not from the sitemap-module for nuxt 2.x

**edit:**
@FairyFanFan  You shared the source code from another repository : https://github.com/nuxt-modules/sitemap/blob/v2.4.24/src/runtime/util/entries.ts#L47

**Comment by FairyFanFan** [comment]
> > > > ```js
> > > > lastmod: new Date().toISOString()
> > > > ```
> > > 
> > > 
> > > As bellow:
> > > ```js
> > > lastmod: new Date().toISOString().substring(0, 10)
> > > ```
> > 
> > 
> > thx, but it will be forced to transfer to iso type, cause so
> > urce code in v2.4.0 are "normaliseDate(e.lastmod)", normalise function force date string transfer to iso type <img width="603" alt="截屏2024-07-29 17 18 33" src=" <img width="650" alt="截屏2024-07-29 17 18 59" src="https://github.com/user-attachments/assets/5 <img alt="截屏2024-07-29 17 19 18" width="688" src="https://private-user-images.githubusercontent.com/12137715/352989500-960abdeb-eaa5-4f61-adff-0a43fa278214.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIyNjY4NTksIm5iZiI6MTcyMjI2NjU1OSwicGF0aCI6Ii8xMjEzNzcxNS8zNTI5ODk1MDAtOTYwYWJkZWItZWFhNS00ZjYxLWFkZmYtMGE0M2ZhMjc4MjE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI5VDE1MjIzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMzYjY3NDRmZGMzMmFkZTk1MjdhY2FmNWVjMzQyNTNjMTZkZjc5OGIxZTdlN2IyZmM3Njg2NDk3NDE5NmFiMGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-AV373juotKxexNZdCZD0bV3zf2FGql19cirBIBumA8"> 44446da-96fe-437e-8f00-8a660d222011"> [https://github.com/user-attachments/assets/f5b30a6c-d76c-4f32-a804-b758d49a5a75">](https://github.com/user-attachments/assets/f5b30a6c-d76c-4f32-a804-b758d49a5a75%22%3E)
> 
> I don't understand your explanation 🤔 This typescript code is not from the sitemap-module for nuxt 2.x
> 
> **edit:** @FairyFanFan You shared the source code from another repository : https://github.com/nuxt-modules/sitemap/blob/v2.4.24/src/runtime/util/entries.ts#L47

when u use @nuxtjs/sitemap v2.4.0, u can check it in package.json that it  depends on sitemap 4.1.1, then u check node_modules/sitemap/dist/lib/sitemap.js,  lastmod is forced to transfer to toISOString()
....
but i fixed it 
my solution is:  downgrade @nuxtjs/sitemap v2.4.0 to v1.3.1,  cause it depends on "sitemap": "^2.2.0",  lastmod in it is not forced to iso type

**Comment by NicoPennec** [comment]
@FairyFanFan According the [sitemap protocol](https://www.sitemaps.org/protocol.html), the `lastmod` date should be in [W3C Datetime](https://www.w3.org/TR/NOTE-datetime) format.
